upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1286,24 +1288,6 @@ func newNodeUpConfigBuilder(cluster *kops.Cluster, assetBuilder *assets.AssetBui
 			}
 		}
 
-		if isMaster || useGossip {
-			protokubeImage[role] = make(map[architectures.Architecture]*nodeup.Image)
-			for _, arch := range architectures.GetSupported() {
-				u, hash, err := ProtokubeImageSource(assetBuilder, arch)
-				if err != nil {
-					return nil, err
-				}
-
-				asset := mirrors.BuildMirroredAsset(u, hash)
-
-				protokubeImage[role][arch] = &nodeup.Image{
-					Name:    kopsbase.DefaultProtokubeImageName(),
-					Sources: asset.Locations,
-					Hash:    asset.Hash.Hex(),
-				}
-			}
-		}
-
```

Comment:
Any special reason why this logic was completely removed? I think we don't want not to download the new binaries on any nodes, even when not needed.

Reply:
I don't see the need of this piece of logic at this point. With the new code we are adding the protokube and channels binary file assets in `addFileAssets` and we are picking up the architecture specific protokube and channels binary in the protokube nodeup code. This logic seems to be adding the image to assets in `newNodeUpConfigBuilder`. I have essentially shifted the place where the asset is added.

Please do correct me if I have missed something here.

Reply:
https://github.com/kubernetes/kops/pull/10574#discussion_r584100010